### ISSUE TITLE
component-selector and directive-selector issues

### ISSUE THREAD

**Issue by PhiLhoSoft** [angular-eslint/angular-eslint#943]
**Description and reproduction of the issue**

I have defined these rules (in the overrides section for *.ts):

```javascript
	'@angular-eslint/component-selector': ['error', { prefix: ['pr'] }],
	'@angular-eslint/directive-selector': ['error', { prefix: ['pr'] }],
```

I believe they worked the first time I tried them, as they reported relevant errors.
Later, I upgraded this package, and eventually found out they were no longer working.
It turns out I have to define all fields, event those marked as optional, even as the rule code seems to provide significant defaults (which, by the way, are not indicated in the rule documentations).

I found out I had to indicate these parameters:

```javascript
	'@angular-eslint/component-selector': ['error', { type: 'element', prefix: ['pr'], style: 'kebab-case' }],
	'@angular-eslint/directive-selector': ['error', { type: 'attribute', prefix: ['pr'], style: 'camelCase' }],
```

They are actually needed, as they are different, but still, there is either (or both?) a problem with default handling or with documentation (should not be marked as optional).

Note: good or bad practice, I don't know, but our projects have some dual directive selectors, like `selector: '[prDrop], pr-drop',`, ie. defining the directive both as attribute and as element… I suppose I have to deactivate the rule for these cases. I wonder if directive-selector can handle both configurations, or if these cases are too marginal to be taken care of.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `13.1.0` |
| `@typescript-eslint/parser`        | `5.15.0` |
| `ESLint`                           | `8.11.0` |
| `node`                             | `14.18.2` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 13.3.0
Node: 14.18.2
Package Manager: npm 7.24.2
OS: win32 x64

Angular: 13.3.0
... cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1303.0
@angular-devkit/build-angular   13.3.0
@angular-devkit/core            13.3.0
@angular-devkit/schematics      13.3.0
@schematics/angular             13.3.0
ng-packagr                      13.3.0
rxjs                            7.5.5
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by Abildtoft** [comment]
Similar issue here. For legacy reasons, we don't use prefixes (yet, working on it) for our component selectors.

So I have to provide:

```
"@angular-eslint/component-selector": [
	"warn",
	{
		"type": ["element", "attribute"],
		"prefix": [""],
		"style": "kebab-case"
	}
],
```

If the prefix property is left out, the rule doesn't appear to be applied.

**Comment by Shivugethub** [comment]
Hi, is the issue still exist, anybody still facing the issue?

**Comment by JamesHenry** [comment]
These rules should be more consistent and reliable as of https://www.npmjs.com/package/angular-eslint/v/21.0.0

`prefix: ''` and `prefix: []` are both acceptable for no prefix, and the prefix value defaults to `app` if not set at all to align with the Angular documentation/CLI's recommendation
